pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -175,7 +179,22 @@ func (m *manager) Start(activePods ActivePodsFunc, podStatusProvider status.PodS
 
 func (m *manager) AddContainer(p *v1.Pod, c *v1.Container, containerID string) error {
 	m.Lock()
-	err := m.policy.AddContainer(m.state, p, c, containerID)
+	// Proactively remove CPUs from init containers that have already run.
+	// They are guaranteed to have run to completion before any other
+	// container is run.
+	for _, initContainer := range p.Spec.InitContainers {
```

Comment:
@klueska we have discussed this part of the patch a couple times and it’s good to get it in. However the PR title / description don’t mention this improvement to how reliably the cpu manager reuses init container guaranteed CPUs.

Reply:
This was added in https://github.com/kubernetes/kubernetes/pull/78762

The current patch just moves the logic up into the CPUManager instead of down in the static policy.

Reply:
Ok, weird — I must have missed the delete part of the diff somehow. All good then!